tests/gnrc_dhcpv6_client_stateless/tests-as-root/01-run.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2018 Freie Universität Berlin
+#               2021 Jan Romann <jan.romann@uni-bremen.de>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import os
+import pexpect
+import random
+import sys
+import time
+
+from scapy.all import AsyncSniffer, sendp, Ether, IPv6, UDP, ICMPv6ND_RS, ICMPv6ND_RA
+from scapy.all import DHCP6_Reply, DHCP6_InfoRequest, ICMPv6NDOptSrcLLAddr, ICMPv6NDOptMTU
+from scapy.all import DHCP6OptClientId, DHCP6OptServerId, ICMPv6NDOptPrefixInfo
+from scapy.all import DUID_LL
+from testrunner import run
+
+try:
+    from scapy.all import DHCP6OptMudUrl
+    mud_option_loaded = True
+except ImportError:
+    from scapy.all import DHCP6OptUnknown
+    DHCP6OptMudUrl = DHCP6OptUnknown
+    mud_option_loaded = False
+
+TIMEOUT = 1
+
+MUD_OPTION_CODE = 112
+MUD_TEST_URL = b'https://example.org'
+
+
+def get_hwaddrs(child):
+    hwaddrs = []
+    child.sendline("ifconfig")
+    child.expect(r"HWaddr:\s+(([A-Fa-f0-9]{2}:?)+)\s")
+    hwaddrs.append(child.match.group(1).lower())
+    if len(hwaddrs[0]) == 5:    # short address
+        res = child.expect([pexpect.TIMEOUT,
+                            r"Long HWaddr:\s+(([A-Fa-f0-9]{2}:?)+)\s"])
+        if res > 0:
+            hwaddrs.append(child.match.group(1).lower())
+    return hwaddrs
+
+
+def start_sniffer(iface, filter, count=None, stop_filter=None):
+    sniffer = AsyncSniffer(
+        iface=iface,
+        filter=filter,
+        count=count,
+        stop_filter=stop_filter,
+    )
+    sniffer.start()
+    return sniffer
+
+
+def wait_for_dhcpv6_pkt(iface, sniffer=None, timeout=5):
+    if sniffer is None:
+        sniffer = start_sniffer(iface, "udp and dst port 547", count=3000)
+    sniffer.join(timeout=timeout)
+    if sniffer.results is None:
+        raise TimeoutError("Sniffing for DHCPv6 traffic timed out")
+    return [p for p in sniffer.results
+            # filter out packets only belonging to stop_filter if it existed
+            if sniffer.kwargs.get("stop_filter") is None or
+            sniffer.kwargs["stop_filter"](p)][-1]
+
+
+def wait_for_icmpv6_pkt(iface, sniffer=None, timeout=15):
```

Comment:
Did not look into the test code in detail, but might it make sense to use a [Scapy Automaton](https://scapy.readthedocs.io/en/latest/advanced_usage.html#first-example) for this?

Reply:
Oh, that looks very promising! I'll try to refactor the test code using an automaton

Reply:
If you want to have an example how to have this interact with `pexpect`/`testutils`, have a look at the [`emcute` test](https://github.com/RIOT-OS/RIOT/blob/master/examples/bindist/README.md) where I also used it to emulate an MQTT-SN broker.

Reply:
I now added a first refactored version using an Automaton :) While the test still works, the might be some room for refinement. Let me know if I should change anything :)